engines/router/missionctl/server/upi/logutils.go

```diff
@@ -0,0 +1,87 @@
+package upi
+
+import (
+	"context"
+	"time"
+
+	"github.com/caraml-dev/turing/engines/router/missionctl/errors"
+	"github.com/caraml-dev/turing/engines/router/missionctl/log"
+	"github.com/caraml-dev/turing/engines/router/missionctl/log/resultlog"
+	upiv1 "github.com/caraml-dev/universal-prediction-interface/gen/go/grpc/caraml/upi/v1"
+	"google.golang.org/grpc/metadata"
+)
+
+type grpcRouterResponse struct {
+	key    string
+	header metadata.MD
+	body   *upiv1.PredictValuesResponse
+	err    string
+}
+
+func logTuringRouterRequestSummary(
+	ctx context.Context,
+	timestamp time.Time,
+	header metadata.MD,
+	body []byte,
+	mcRespCh <-chan grpcRouterResponse,
+) {
+
+	// Create a new TuringResultLogEntry record with the context and request info
+	logEntry := resultlog.NewTuringResultLogEntry(ctx, timestamp, header, string(body))
```

Comment:
What does `string(body)` produces? I think we should convert it into json equivalent proto for now.